waku/v2/protocol/waku_store/rpc.nim

```diff
@@ -112,55 +112,53 @@ proc toAPI*(rpc: PagingIndexRPC): HistoryCursor =
 
 
 proc toRPC*(query: HistoryQuery): HistoryQueryRPC =
-  let 
-    contentFilters = query.contentTopics.mapIt(HistoryContentFilterRPC(contentTopic: it))
-
-    pubsubTopic = query.pubsubTopic.get(default(string))
-  
-    pageSize = query.pageSize
-
-    cursor = query.cursor.get(default(HistoryCursor)).toRPC()
+  var rpc = HistoryQueryRPC()
+
+  rpc.contentFilters = query.contentTopics.mapIt(HistoryContentFilterRPC(contentTopic: it))
+
+  rpc.pubsubTopic = query.pubsubTopic
+
+  rpc.pagingInfo = block:
+      if query.cursor.isNone() and 
+         query.pageSize == default(type query.pageSize) and 
+         query.ascending == default(type query.ascending): 
+        none(PagingInfoRPC)
+      else:
+        let
+          pageSize = some(query.pageSize)
+          cursor = query.cursor.map(toRPC)
+          direction = if query.ascending: some(PagingDirectionRPC.FORWARD)
+                      else: some(PagingDirectionRPC.BACKWARD)
+        some(PagingInfoRPC(
+          pageSize: pageSize,
+          cursor: cursor,
+          direction: direction
+        ))
+      
+  rpc.startTime = query.startTime
+  rpc.endTime = query.endTime
+
+  rpc
 
-    direction = if query.ascending: PagingDirectionRPC.FORWARD
-                else: PagingDirectionRPC.BACKWARD
-
-    startTime = query.startTime.get(default(Timestamp))
-    
-    endTime = query.endTime.get(default(Timestamp))
-
-  HistoryQueryRPC(
-    contentFilters: contentFilters,
-    pubsubTopic: pubsubTopic,
-    pagingInfo: PagingInfoRPC(
-      pageSize: pageSize,
-      cursor: cursor,
-      direction: direction
-    ),
-    startTime: startTime,
-    endTime: endTime
-  )
 
 proc toAPI*(rpc: HistoryQueryRPC): HistoryQuery =
   let 
-    pubsubTopic = if rpc.pubsubTopic == default(string): none(PubsubTopic)
-                  else: some(rpc.pubsubTopic)
+    pubsubTopic = rpc.pubsubTopic
     
     contentTopics = rpc.contentFilters.mapIt(it.contentTopic)
 
-    cursor = if rpc.pagingInfo == default(PagingInfoRPC) or rpc.pagingInfo.cursor == default(PagingIndexRPC): none(HistoryCursor)
-             else: some(rpc.pagingInfo.cursor.toAPI())
+    cursor = if rpc.pagingInfo.isNone() or rpc.pagingInfo.get().cursor.isNone(): none(HistoryCursor)
+             else: rpc.pagingInfo.get().cursor.map(toAPI)
 
-    startTime = if rpc.startTime == default(Timestamp): none(Timestamp)
-                else: some(rpc.startTime)
+    startTime = rpc.startTime
 
-    endTime = if rpc.endTime == default(Timestamp): none(Timestamp)
-              else: some(rpc.endTime)
+    endTime = rpc.endTime
     
-    pageSize = if rpc.pagingInfo == default(PagingInfoRPC): 0.uint64
-               else: rpc.pagingInfo.pageSize
+    pageSize = if rpc.pagingInfo.isNone() or rpc.pagingInfo.get().pageSize.isNone(): 0'u64
```

Comment:
Is there a better way to check if a nested option isNone, i.e a proc/template like -
```nim
nestedFieldIsNone(pagingInfo, pageSize)
```
no real opinion on this, just curious :)

Reply:
I know in other languages it exists, like Kotlin's `?` operator. I don't know about Nim.

In any case, here, the solution is to simplify and relocate the protobuf message fields.

Reply:
Sounds good!